upp/classes/components.py

```diff
@@ -82,7 +91,12 @@ def check_num_jets(
             log.info(f"Estimated {available:,} {self} jets available - {num_req:,} requested")
 
     def get_auto_sampling_frac(self, num_jets, cuts=None, silent=False):
-        total = self.reader.estimate_available_jets(cuts, self.num_jets_estimate)
+        if self.num_jets_estimate_num <= 0:
+            self.setup_reader(**self.reader_replicate_kwargs)
+            total = self.reader.estimate_available_jets(cuts, None)
+        else:
+            self.setup_reader(**self.reader_replicate_kwargs)
+            total = self.reader.estimate_available_jets(cuts, self.num_jets_estimate_num)
```

Comment:
you can condense this with something like

```
arg = self.num_jets_estimate_num if ... else ....
self.setup_reader(...)
total = self.reader.estimate_available_jets(cuts, arg)
```

Reply:
Thanks for the suggestion, did it in thee corresponding commit